### PR TITLE
Data dictionary valid values

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
Spec: "External Release 2" in the [spec's Release Sequence](https://docs.google.com/document/d/1BPhdopyt07xNVv1uZ000094j25otKm6Deut1yXRSsyo/edit#heading=h.tlpffswyn3ey)

Showing valid values in the data dictionary:
Screen Shot 2021-11-24 at 1 27 26 PM

Showing valid values while importing cases:
Screen Shot 2021-11-24 at 1 40 43 PM

Showing warnings after import:
Screen Shot 2021-11-24 at 1 28 04 PM

## Feature Flag
Validate data per data dictionary definitions during case import

### Safety story
This has tests and has gone through QA. Issues should be limited to flagged domains and to the affected features (data dictionary & case import).

### Automated test coverage

There's decent test coverage in this PR.

### QA Plan

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-3436)

### Migrations

The migration in this PR just adds a new model.

- [x] The migrations in this code can be safely applied first independently of the code

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by orangejenny** [comment]
All of this, except the final two cleanup commits, has been reviewed in other PRs.

**Review by AddisonDunn — APPROVED** [review]
Obviously, I only reviewed the last 2 commits, which look like simple enough changes to me.

File: corehq/apps/data_dictionary/tests/utils.py [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
     prop_list = prop_list or []
+    allowed_values = allowed_values or {}
     CaseType.get_or_create(domain, case_type_name)
     for prop_name, data_type in prop_list:
         prop = CaseProperty.get_or_create(prop_name, case_type_name, domain)
         prop.data_type = data_type
         prop.save()
+        if prop_name in allowed_values:
+            for value in allowed_values[prop_name]:
+                CasePropertyAllowedValue.objects.create(case_property=prop, allowed_value=value)
```
Comment (snopoke) [inline corehq/apps/data_dictionary/tests/utils.py:14]: can this be done in bulk? Also should this be a `get_or_create`?

Comment (orangejenny) [reply corehq/apps/data_dictionary/tests/utils.py:14]: yeah, `get_or_create` makes sense. I could go either way on the bulk save, as there aren't likely to be that many values. @kmtracey could you update this?

Comment (orangejenny) [reply corehq/apps/data_dictionary/tests/utils.py]: Updated in https://github.com/dimagi/commcare-hq/pull/30786

**Pull request merged**